### ISSUE TITLE
status(object)[O_CALLOUTS] empty array

### ISSUE THREAD

**Issue by nyankers** [dworkin/dgd#60]
Except when `obj` is `this_object()`, `status(obj)[O_CALLOUTS]` always seems to return an empty array.

I'm not 100% certain but I think this is due to `config.cpp:1971`:

```
	    a = data->listCallouts(obj->dataspace());
```

I think this is backwards, as `listCallouts()` assumes the callouts are in the dataspace it's being called on, and just use the provided dataspace to create arrays and such. Changing this line to the below seems to resolve the issue:

```
	    a = obj->dataspace()->listCallouts(data);
```

Though I'm not sure if there are any issues with this change (e.g. will `obj->dataspace()` always return a dataspace, etc.), so unless you want me to investigate further, I'll leave the actual change to you. ^^

**Comment by dworkin** [comment]
Your bug report and analysis are correct, fixed in 4a01f59a03b63b3d87873fe87cae0e87e07bd853.

**Comment by nyankers** [comment]
Belatedly, looks like it works so far, thanks! ^^
